pkg/services/store/entity/server/service.go

```diff
@@ -147,3 +158,36 @@ func (s *service) running(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (s *service) initInstrumentation(ctx context.Context) *http.Server {
+	// Only start up instrumentation server when storage-server is the sole target module
+	if len(s.cfg.Target) == 1 && s.cfg.Target[0] == modules.StorageServer {
+		return s.setupInstrumentationServer(ctx)
+	}
```

Comment:
Just to confirm, this means we won't be getting metrics in multi-target mode ?  Maybe add a TODO?

Reply:
it seems like this would be needed any time we're not starting the main http server, right?

Reply:
@chaudyg @DanCech  Yeah. I wasn't sure what behaviour we want in multi-target mode (or if that's actually a use case?). Might be better to start the instrumentation server when it is a single target and not `all` or `core`?

Reply:
I guess the other thing is, do we want the service responsible for starting the metrics server? If we end up running multiple targets, it might be awkward to check if there's already one running.

Reply:
I refactored it so that it runs the instrumentation server module if the target isnt `all` or `core`. I gets registered an invisible module, but I'm not sure what different that makes.